sdk/php/runtime/template/README.md

```diff
@@ -311,3 +311,48 @@ class Example
      // ...
 }
 ```
+
+#### Directories and Files
+
+Directories and Files have some additional metadata that can be specified
+
+##### Defaults
+
+Specifying a default value for a `Directory` or `File` isn't easily possible. 
+Instead you can specify a `DefaultPath` attribute with an absolute or relative path.
+If relative then the path is relative to the `source` specified in the `dagger.json` file.
+
+This tells Dagger; if no argument is specified, fetch the `Directory` or `File` from the `DefaultPath`.
+
+The following example shows how you would get the parent directory of your `source`:
+
+```php
+#[DaggerFunction]
+public function myDaggerFunction(
+    #[DefaultPath('../')]
+    Directory $dir,
+): Container {
+    // ...
+}
+```
+
+##### Ignore
+
+For Directories only, an `Ignore` Attribute to specify what files|directories should be ignored.
+This is done by providing an array of strings.
+Each string follows the [`.gitignore` syntax](https://git-scm.com/docs/gitignore/en): 
+prefixing a path with `!` will actually include it instead of ignoring it.
```

Comment:
It's a bit more subtle, when prefix a path with `!`, you actually negate a precedent ignore.

For example: `["go/*", "!go/main.go"]` will ignore everything in the `go` dir except `go/main.go`.
So if you just do `["!main.go"]`, nothing is actually ignored at all.

Also keep in mind that the order of the ignore is important.
If you do `["!main.go", "**"]`, everything will be ignored since the main.go is included then exclude.

Reply:
I pinched and reworded this bit, so this may want a bit more detail too:

https://github.com/dagger/dagger/blob/a2d4b1f0cd11a0affde61836b980f4462174e92e/docs/current_docs/manuals/developer/functions.mdx?plain=1#L435

I can update this at the same time to match what you've just taught me if you'd like?

Reply:
Yeah I think that's better to focus on usage in this doc, and check for gitignore page is some1 need more conceptual info